bin/console.picocom

```diff
@@ -22,4 +22,21 @@ if [ -z "${console_baud}" ]; then
     console_baud=115200
 fi
 
+# Some boards have a UART built into them and it doesn't appear until the
```

Comment:
I previously thought this couldn't work, but now that I look at u_boot_console_exec_attach.py I see that it will.

The test/py console first flashes the device *then* opens that console to the device to interact with it. Thus, waiting for the console to appear won't prevent the flashing process from operating, and any power cycles during the flashing process won't disconnect the console connection since none has been established yet. TBH, I think this is a bug in test/py; the console connection should be established before flashing occurs so that any console output that occurs during flashing can't be captured with the current scheme. However, this bug is good for your use-case since it allows for consoles that appear/disappear during flashing.

FYI, my U-Boot build scripts do the following to at least prevent U-Boot from auto-booting before the host can start to interact with it. This doesn't prevent loss of the start of the captured console output, but does at least give the active interaction more time to start.
```
echo 'CONFIG_BOOTDELAY=15' >> "src/u-boot/configs/${u_boot_board}_defconfig"
```